dashboard/tasks.py

```diff
@@ -68,18 +69,34 @@ def release_lock():
         return cache.delete(lock_id)
 
     if acquire_lock():
-        enrolled_students = CachedEnrollment.objects.filter(
-            last_request__lt=datetime.utcnow() - timedelta(hours=MAX_HRS_MAIN_TASK)
-        ).values_list('user', flat=True)
+        refresh_time_limit = datetime.utcnow() - timedelta(hours=MAX_HRS_MAIN_TASK)
+
+        users_expired_enroll = CachedEnrollment.objects.filter(
+            last_request__lt=refresh_time_limit
+        ).values_list('user', flat=True).distinct()
+
+        users_expired_certs = CachedCertificate.objects.filter(
+            last_request__lt=refresh_time_limit
+        ).values_list('user', flat=True).distinct()
+
+        users_expired_grades = CachedCurrentGrade.objects.filter(
+            last_request__lt=refresh_time_limit
+        ).values_list('user', flat=True).distinct()
 
-        certificated_students = CachedCertificate.objects.filter(
-            last_request__lt=datetime.utcnow() - timedelta(hours=MAX_HRS_MAIN_TASK)
-        ).values_list('user', flat=True)
+        users_expired_cache = list(
```

Comment:
We don't need to convert to a `list` here, Django will accept a set for the `id__in` parameter